### ISSUE TITLE
Duplicate GlobalKey detected in widget tree.

### ISSUE THREAD

**Issue by Trung15010802** [salim-lachdhaf/searchable_dropdown#672]
**Describe the bug**
Duplicate GlobalKey detected in widget tree.

**To Reproduce**
If we click on the dropdown many times very quickly the error will appear.

**Expected behavior**
Duplicate GlobalKey detected in widget tree.

**Code**
```dart
import 'package:dropdown_search/dropdown_search.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: SizedBox(
          width: 300,
          child: DropdownSearch(
            items: (filter, loadProps) => [1, 2, 3, 4, 5],
          ),
        ),
      ),
    );
  }
}

```

**Screenshots**

https://github.com/user-attachments/assets/a3f6d2e5-731a-44bb-8f1e-9ddae533bbc3

**Comment by salim-lachdhaf** [comment]
Thank you for the demo, i will check it

**Comment by salim-lachdhaf** [comment]
Finally, I identified the issue. The problem is actually caused by reopening the popup while the previous popup hasn’t been fully dismissed yet. You can verify this by clicking very quickly on another item that is not at the top of the dropdown; there won't be any issue in that case.

a fix will be released in the next days.

**Comment by salim-lachdhaf** [comment]
fixed in 6.0.1, it will be released today
